libs/testdiff/testdiff.go

```diff
@@ -0,0 +1,91 @@
+package testdiff
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/databricks/cli/internal/testutil"
+	"github.com/hexops/gotextdiff"
+	"github.com/hexops/gotextdiff/myers"
+	"github.com/hexops/gotextdiff/span"
+	"github.com/stretchr/testify/assert"
+	"github.com/wI2L/jsondiff"
+)
+
+func UnifiedDiff(filename1, filename2, s1, s2 string) string {
+	edits := myers.ComputeEdits(span.URIFromPath(filename1), s1, s2)
+	return fmt.Sprint(gotextdiff.ToUnified(filename1, filename2, s1, edits))
+}
+
+func AssertEqualTexts(t testutil.TestingT, filename1, filename2, expected, out string) {
+	if len(out) < 1000 && len(expected) < 1000 {
+		// This shows full strings + diff which could be useful when debugging newlines
+		assert.Equal(t, expected, out)
+	} else {
+		// only show diff for large texts
+		diff := UnifiedDiff(filename1, filename2, expected, out)
+		t.Errorf("Diff:\n" + diff)
+	}
+}
+
+func AssertEqualJQ(t *testing.T, expectedName, outName, expected, out string, ignorePaths []string) {
+	patch, err := jsondiff.CompareJSON([]byte(expected), []byte(out))
```

Comment:
Have you explored using `dyn` libraries instead https://github.com/databricks/cli/blob/main/libs/dyn/jsonloader/json.go ? It allows for loading and similar comparison of JSON payloads? If yes, what are benefits of using `jsondiff` instead?

Reply:
No and I don't see comparable API there. It would be some effort to evaluate that option, let's do as a follow up.